src/test/java/net/openhft/chronicle/queue/TestDeleteQueueFile.java

```diff
@@ -192,38 +198,279 @@ public void firstAndLastIndexAreRefreshedAfterForceRefreshInterval() throws IOEx
+                    Files.delete(Paths.get(rollCycleDetails.filename));
+                    deletedUpTo++;
+                }
+                queueWithCycleDetails.queue.refreshDirectoryListing();
+                Jvm.pause(1_000);
+            }
+        } catch (IOException e) {
+            Jvm.error().on(TestDeleteQueueFile.class, "Error occurred", e);
+        }
+    }
+
+    private void deleteAllRollCyclesInRandomOrder(QueueWithCycleDetails queueWithCycleDetails) {
+        try {
+            int numberOfCycles = queueWithCycleDetails.rollCycles.size();
+            int deleted = 0;
+            while (queueWithCycleDetails.rollCycles.size() > 1) {
+                // Don't delete the first cycle, we can't deal with that yet
+                final int index = ThreadLocalRandom.current().nextInt(1, queueWithCycleDetails.rollCycles.size());
+                final RollCycleDetails rollCycleDetails = queueWithCycleDetails.rollCycles.remove(index);
+                deleted++;
+                Jvm.startup().on(TestDeleteQueueFile.class, "Deleting " + rollCycleDetails.rollCycle + ": " + rollCycleDetails.filename + " (" + deleted + "/" + numberOfCycles + "), firstIndex=" + toHexString(rollCycleDetails.firstIndex) + ", lastIndex=" + toHexString(rollCycleDetails.lastIndex));
+                Files.delete(Paths.get(rollCycleDetails.filename));
+                queueWithCycleDetails.queue.refreshDirectoryListing();
+                Jvm.pause(20);
+            }
+        } catch (IOException e) {
+            Jvm.error().on(TestDeleteQueueFile.class, "Error occurred", e);
+        }
+    }
+
+    @Test
+    public void deletingRandomRollCyclesChaosTest() throws InterruptedException {
+        ignoreException("The current cycle seems to have been deleted from under the queue, scanning to find the next remaining cycle");
+        final int numberOfCycles = 300;
+        final AtomicBoolean running = new AtomicBoolean(true);
+        try (QueueWithCycleDetails queueWithCycleDetails = createQueueWithNRollCycles(numberOfCycles, null)) {
+
+            Thread forwardTailerThread = new Thread(new QueueTailer(running, queueWithCycleDetails, TailerDirection.FORWARD));
+            Thread backwardTailerThread = new Thread(new QueueTailer(running, queueWithCycleDetails, TailerDirection.BACKWARD));
+
+            Thread deleteRandomCyclesThread = new Thread(() -> deleteAllRollCyclesInRandomOrder(queueWithCycleDetails));
+
+            backwardTailerThread.start();
+            forwardTailerThread.start();
+            deleteRandomCyclesThread.start();
+            deleteRandomCyclesThread.join();
+            running.set(false);
+            backwardTailerThread.join();
+            forwardTailerThread.join();
+        }
+    }
```

Comment:
This test randomly deletes cycles at any point in the queue. It probably doesn't mirror any kind of sensible archiving strategy, it was more to add confidence that Tailers are still able to traverse under chaotic conditions.

Again, while that's occurring, we have two threads continually scanning over the queue in both directions and validation that they saw all the remaining roll cycles. (this is a bit hard to check so we ensure they saw all the roll cycles not deleted when they reached the start/end of the queue).